backend/mlarchive/authbackend/oidc.py

```diff
@@ -18,3 +21,32 @@ def get_logout_url(request):
     endpoint = settings.OIDC_OP_X_END_SESSION_ENDPOINT
     url = f'{endpoint}?post_logout_redirect_uri={redirect}&id_token_hint={token}'
     return url
+
+
+class CustomOIDCBackend(OIDCAuthenticationBackend):
+
+    def create_user(self, claims):
+        user = super().create_user(claims)
+        # extra logic
+        self.add_list_membership(user, claims)
+        return user
+
+    def update_user(self, user, claims):
+        user = super().update_user(user, claims)
+        # extra logic
+        self.add_list_membership(user, claims)
+        return user
+
+    def add_list_membership(self, user, claims):
+        '''Get all known emails for user from Datatracker. If they have a new email
+        create UserEmail and add any EmailList.members relations as needed
+        '''
+        # get all emails from claims
+        # not doing this initially, re-evaluate after new authenication system deployed
+        # known_emails = claims.get('emails')
+        known_emails = get_known_emails(user.email)
```

Comment:
This call may fail or time out separately from the OIDC flow. At the least, need to catch such a failure and handle it gracefully.

I think it'd be best if that didn't break access to the mailarchive, even if that means going ahead with stale email address information.

Reply:
Added generic try / except to eliminate errors during login